Ness Kit touchups + Lucas Tweaks

O1yyy opened the pull request:
assets: 

[pk_assets.zip](https://github.com/HDR-Development/HewDraw-Remix/files/12286956/pk_assets.zip)


## BIG thank you to Riah for cleaning up most of the kit
### please refer to https://github.com/HDR-Development/HewDraw-Remix/pull/1770 for all of the Ness Hitbox changes
## also thank you jaccob101 for the Lucas Magnet Hitlag tweaks ~~and Ness DThrow FAF edits,~~ an old pr is linked here: https://github.com/HDR-Development/HewDraw-Remix/pull/1872

high level interpretation of my changes: 

# ness

### pkthunder
- pkt2 blue screen removed
- pkt1's hover is much better, negligible fall acceleration now
- turning angle on pkt1 ball buffed, should make chosing angles feel a little better
- you can drift much sooner after pkt2 ends
- fixed wall collision, no longer dies on walls
- now grabs ledge behind him while at peak pkt2 (nutty underside angles are now possible)
- pkthunder on both kids now can edge slipoff on startup and edge cancels at the peak
- decreased initial speed: 3.7 -> 3.5 to account for wall jump

### added a wall jump
- decreased vanilla jump height
- passive_wall_jump_y_speed: 2.5 ->1.9


### ~~pkfire~~
~~- pkfire no longer recoils~~
- nvm job did it

### magnet
- fixed the jc frame, JC on F8 (hitbox still on F7)
- hitlag mult 1.0x -> 0.5x (JC after 3 frames of hitlag: F10)
- removed set kb & 361 angle, switched to kbg/bkb, lessened bkb and gave it kbg
  - angle 361 -> 55
  - FKB: 70 -> 0
  - KBG: 3 -> 135
  - BKB: 20 -> 10
- cleaned up effects
- added a helper hitbox to help send opponents forward more reliably

![image](https://github.com/HDR-Development/HewDraw-Remix/assets/19886440/1c94f6ab-676e-4496-b0da-0bc861953b9a)

### nair
- pulled visual into the torso to better match the hitboxes
- Early hit activity F5-F12 -> F5->F9
- Late hit activity F13-F16 -> F9 -> F16
- Late hit KBG 100 -> 90
- Late hit BKB 15 -> 25

### Dair:
- Late hit activity F24-F30 -> F24-F29

### fair
- multihits
  - angle 365 -> 361
  - tweaked kb to compensate for pm esque behaviour
  - cleaned up acmd so the gaps between hits are consistent
  - added a helper hitbox covering the lenght of his body
    - same stats as main psi hitbox
  
https://github.com/HDR-Development/HewDraw-Remix/assets/19886440/09d49494-2cf3-49fb-9c36-98f2a4cbe7d8


### usmash
- fixed double hit usmash
- fixed no hitbox when uncharged 

### dsmash
- cleaned up hitboxes to fix blindspots, comparable to pm now

### dthrow
~~- FAF 50 -> 45~~
- kbg: 85 -> 65
- bkb: 70 -> 80

### jab 1
- Angle changed. 361 -> 70 (jablock still 361)
- Damage 2.0 -> 3.0
- KBG 15/25 -> 25
- Hitlag mul 1.7 -> 1.0
- Adjusted G/A Priorities for jablock/normal hitbox.

### jab 2
- Angle changes 361 -> 55 (jablock still 361)
- Damage 2.0 -> 3.0
- KBG 20/25 -> 25.
- Jablock hitlag mul 1.2 -> 1.0
- Adjusted G/A Priorities for jablock/normal hitbox.

### dash attack
- Single hit 1: damage 7.0 -> 8.0

- Hit 1 ~
  - damage 4.0/3.0 -> 4.0
  - Angle changes 0/45 -> 0/361
  - SDI muls 1.0 -> 0.5

- Hit 2 ~
  - SDI Mul 1.0 -> 0.5

- Hit 3 
  - Hitlag mul 1.5 -> 1.3

### dtilt
- Dtilt adjusted to P+
  - angle 361 -> 83
  - inner hitbox damage 4.5% -> 3.0%, does same damage allover
  - KBG: 3 -> 135
  - BKB: 20 -> 10
  - hitlag mult: 1.0 -> 0.85
  - sdi mult: 1.0 -> 0.5

# lucas
### zair
- active on frames f9-22 -->f7-10
- damage: 4% --> 8%
- angle: 70
- kbg: 70
- bkb: 30
- hitlag mult: 1.3x
- ledge box check: frame 7 --> 6

https://github.com/HDR-Development/HewDraw-Remix/assets/19886440/6d1e3828-26cc-41cc-93fb-dcfde34bfb68

### up b
- multihits
  - now interpolate properly
- id 0: (front head hitbox)
  - damage: 1.0% -> 2.0%
  - angle: 365 -> 366
  - kbg: 100 -> 130
  - fkb: 180 -> 50
  - sdi mult: 1.5x -> 0,5x
- id 1: (back end hitbox)
  - damage: 2.0% -> 1.0% (encourage sdi attempts)
  - same stats as id 1
- finisher
  - matched hitbox layout to multihits
  - size of id 0: 12.5 -> 8.5

![image](https://github.com/HDR-Development/HewDraw-Remix/assets/19886440/a35b610e-9687-4432-925c-d812d1b0acda)

* NOT TO SCALE, but should give a rough idea
* these changes end up not doing much to sdi'ing the move (basically its still easy to get out with sdi and di in), since lucas is moving forward relatively quickly, they only aim to make the move feel more consistent

### charge burst freeze rest
- no longer a rest, you need to confirm off of something to kill with it
  - kbg 115 -> 60
  - bkb 50 -> 60

### pkfreeze side b
- removed ability to drift and fast fall
- to reliably rapid fire/use it, you need to use your dj to djc and land at a similar timing
  - this change is conpensated by the fact that djc side b is much more accessible in this engine with both procons and button remaps available

https://github.com/HDR-Development/HewDraw-Remix/assets/19886440/42851d13-6aaf-43b2-a6e0-754206c0624e

### dtilt
- pushed active frames from f3-4 -> f5-6
- added another sourspot hitbox to cover the gap between the two hitbox a little better
  - concequently makes sweetspot harder to get
  
![2023090102133600-0E7DF678130F4F0FA2C88AE72B47AFDF](https://github.com/HDR-Development/HewDraw-Remix/assets/19886440/f7cc158a-fb4c-4c1d-93d2-6eaaa0b45d90)

### nair
- cleaned up the sound script

### magnet
- (from jaccob101) Matches Magnet hitlag to p+ for input parity. Now should be 1 hitlag on the hold hitbox, and 4 frames on the release hitbox. Previously the release had the 2 frames of hitlag and the hold hitbox had 5 frames. Counting the frame it hits.
  - JC on hitlag: F11
### double jump
- shifted movement arcs back 2 frames inorder to facilitate perfect djc's and general djc feel
  - still has some unsolved quirks, issues documented in https://github.com/HDR-Development/HewDraw-Remix/issues/1888

### params
- air_brake_y 0.023 -> 0.0225
  -  noticed this value wasn't matching pm's, maybe helps him fall a tiny tad faster? i cant tell, probably negligible
- param_pkthunder.speed: 1.5 -> 1.65
  - gave a slight speed increase to the pkt1 ball projectile to help chose angles better

Riah321 committed:
Initial Changes

Riah321 committed:
8 Frames?

Riah321 committed:
Finishing touches and up air effects.

Riah321 committed:
Merge remote-tracking branch 'upstream/dev' into ness-throw-adjustments

Riah321 committed:
more adjustments

Riah321 committed:
Dthrow: 70 KBG -> 85 KBG

Riah321 committed:
Fsmash no longer a warcrime

O1yyy committed:
Ness Misc Fixups

high level interpretation of the changes:

### pkthunder
- pkt2 blue screen removed
- pkt1's hover is much better, negligible fall acceleration now
- turning angle on pkt1 ball buffed, should make chosing angles feel a little better
- you can drift much sooner after pkt2 ends
- fixed wall collision, no longer dies on walls
- now grabs ledge behind him while at peak pkt2 (nutty underside angles are now possible)
### pkfire
- pkfire no longer recoils

O1yyy committed:
temp hotfix cuz i suc

help

O1yyy committed:
Merge pull request #1770 from Riah321/ness-throw-adjustments

Ness Adjustments and effects cleanup.

O1yyy committed:
Backroom Feedback + Fixed Mag JC (?)

- shine still had old weird frame calls, now properly reflects status_frame
  - JC on F9
- reverted nair, matched effects to hitboxes
- reverted down b (ngl i think it was just meant to clean up the hitbox call but w/ver)
- mag effect clean up

O1yyy committed:
fix usmash, cleanup dsmash

- removed double hit usmash
- cleaned up dsmash's hitboxes
 - no longer has blindspots

O1yyy commented:
i dont THINK i need `rom_watch.txt` as part of this pr, if anyone knows what's explicitly needed to declare a new `hdr.xml`, please request the changes

otherwise yeah this is ready for review

O1yyy committed:
mag fixes

- woops i forgot to reset the rehit rate
- back to normal

O1yyy committed:
fix jab pullin + down b stall prevention (attempt)

also disables up b canceling like we did for lucas

O1yyy committed:
Merge branch 'dev' into ness-fixups

O1yyy committed:
update motion.rs

- allows dj arcs to inheret dj anims more naturally

O1yyy committed:
lucas magnet hitlag fixes

- thank you jaccob101!
- matches hitlag to pm's

O1yyy committed:
ness dthrow FAF tweak

- missed this from Riah's pr
- FAF 50 -> 45

O1yyy committed:
Merge branch 'dev' into ness-fixups

O1yyy committed:
param fixes

- lucas:
air_brake_y 0.023 -> 0.0225 (maybe falls a lil faster? iuno can't tell)
- ness
removed penetration_frame edit, accidental left over from lucas params

O1yyy commented:
i left in an unfinished stall prevention script in ness's `opff.rs`, if anyone with a bigger brain wants to help decipher the logic to make it work i will put it for both ness and lucas

O1yyy committed:
pm esque fair + mag tweaks

magnet
- removed set kb, switched to kbg/bkb and gave it kbg
  - fkb 70 -> bkb 70
  - kbg 35
- added a helper hitbox to help send opponents forward more reliably
fair
- multihits
  - angle 365 -> 361
  - tweaked kb to compensate for pm esque behaviour

O1yyy committed:
fix magnet jc for good

moved the jump cancel logic into its own function, and rewrote the logic to only be true if you entered game_speciallw_hold for > 0 frames

O1yyy committed:
misc. touchups

ness
- mag helper box repositioned
- cleaned up mag effects
- cleaned up nair effects
lucas
- fixed mag hitlag
  - JC on hitlag: F?? -> F11
- gave a slight speed increase to the pkt1 ball projectile to help chose angles better

O1yyy committed:
Merge branch 'dev' into ness-fixups

O1yyy committed:
pkthunder edge slipoffs

- pkthunder on both kids now can edge slipoff on startup and endlag
  - still goes through all endlag and does NOT edge cancel
- fixed hitlag mult on ness magnet to match pm muscle memory

O1yyy committed:
Merge branch 'dev' into ness-fixups

WuBoytH commented on fighters/ness/src/acmd/specials.rs:
```diff
@@ -92,16 +124,17 @@ unsafe fn special_lw_hold_game(fighter: &mut L2CAgentBase) {
         if is_excute(fighter) {
-            ATTACK(fighter, 0, 0, Hash40::new("top"), 5.5, 361, 100, 70, 0, 8.5, 0.0, 6.5, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_POS, false, 0, 0.0, 10, false, false, false, false, false, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATT ...
-        }
+            ATTACK(fighter, 0, 0, Hash40::new("top"), 5.5, 361, 35, 0, 70, 2.5, 0.0, 6.5, 2.0, None, None, None, 0.5, 1.0, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_F, false, 0, 0.0, 20, false, false, false, false, false, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_elec"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_MAG ...
```
Remove the 20 rehit rate on these hitboxes, I know it literally doesn't matter but like it does nothing

H3ra-57 commented on utils/agent_params.txt:
```diff
@@ -32,4 +32,6 @@ ganon: fighter/ganon/param/hdr.prc
 miigunner: fighter/miigunner/param/hdr.prc
-falco: fighter/falco/param/hdr.prc
\ No newline at end of file
+miigunner: fighter/miigunner/param/hdr.prc
```
remove extra mii gunner

H3ra-57 reviewed (dismissed):
make sure `rom_watch.txt` is actually up to date

hdr-balance-backroom reviewed (dismissed):
- Document all changes, some are missing such as dthrow's angle and knockback change
- Revert dthrow FAF, readjust knockback to maintain relatively similar combo ability

O1yyy committed:
Merge branch 'dev' into ness-fixups

O1yyy committed:
backroom feedback

description also updated

O1yyy commented:
> * Document all changes, some are missing such as dthrow's angle and knockback change
> 
>     * Revert dthrow FAF, readjust knockback to maintain relatively similar combo ability

so about this, the dthrow angle was from riah's pr, which is documented there and i mention at the very top, any and all hitbox changes are from that pr, my pr only does what i note down to reduce redundancy

backroom looked at that pr in its completed state, dthrow angle and kb included, and this is all you requested that time:

![image](https://github.com/HDR-Development/HewDraw-Remix/assets/19886440/3a511de5-920d-45b0-92fd-122c09769f61)

this stuff i already did so thats taken care of 

so, besides reverting dthrow FAF (and the kb which riah already adjusted), all of my changes are documented, and any other solid number changes are back in https://github.com/HDR-Development/HewDraw-Remix/pull/1770 here

O1yyy committed:
pm lucas zair

- active on frame 9 - 22 --> 7 - 10
- damage: 8%
- angle: 70
- kbg // bkb: 70 // 30
- hitlag mult: 1.3x

O1yyy committed:
Merge branch 'dev' into ness-fixups

O1yyy committed:
the shut up patch

he shuts up more now
- lowered the chances ness does his one 1 (1) (one) damage voiceline

O1yyy committed:
Merge branch 'dev' into ness-fixups

O1yyy committed:
Merge branch 'dev' into ness-fixups

O1yyy committed:
more lucas touchups

mostly nerfs

charge burst freeze rest
- no longer a rest, you need to confirm off of something to kill with it
- kbg 115 -> 60
- bkb 50 -> 60

pkfreeze side b
- removed ability to drift and fast fall
- to reliably rapid fire, you need to use your dj to djc and land at a similar timing
- this change is conpensated by the fact that djc side b is much more accessible in this engine with both procons and button remaps available

dtilt
- pushed active frames from f3-4 -> f5-6
- added another sourspot hitbox to cover the gap between the two hitbox a little better
  - concequently makes sweetspot harder to get

nair
- cleaned up the sound script

O1yyy commented:
ready for review again, will refrain from touching it so that it can get into a beta

O1yyy committed:
ness touchups

Adjusted Jab to be more in line with other jabs.
Further adjusted dash attack's ability to connect all three hits.
Adjusted Nair to increase utility as a combo tool
Adjusted dair to reduce moments where the attack would connect after it appeared to be over.

Jab 1:
Angle changed. 361 -> 70 (jablock still 361)
Damage 2.0 -> 3.0
KBG 15/25 -> 25
Hitlag mul 1.7 -> 1.0 (1.7 -> 1.0 for jablock hitbox)
Adjusted G/A Priorities for jablock/normal hitbox.

Jab 2:
Angle changes 361 -> 55 (jablock still 361)
Damage 2.0 -> 3.0
KBG 20/25 -> 25.
Jablock hitlag mul 1.2 -> 1.0
Adjusted G/A Priorities for jablock/normal hitbox.

Dash Attack:
Single hit 1: damage 7.0 -> 8.0

Hit 1 ~
damage 4.0/3.0 -> 4.0
Angle changes 0/45 -> 0/361
SDI muls 1.0 -> 0.5

Hit 2 ~
SDI Mul 1.0 -> 0.5

Hit 3 ~ Hitlag mul 1.5 -> 1.3

Nair:
Early hit activity F5-F12 -> F5->F9
Late hit activity F13-F16 -> F9 -> F16
Late hit KBG 100 -> 90
Late hit BKB 15 -> 25

Dair:
Late hit activity F24-F30 -> F24-F29

SuddyN committed:
rewrite ground module correct hook for fun

O1yyy committed:
Merge branch 'dev' into ness-fixups

O1yyy committed:
Merge branch 'ness-fixups' of https://github.com/HDR-Development/HewDraw-Remix into ness-fixups

O1yyy committed:
fix pkt2 edge slipoffs + jab lock fix

- edge slipping while at the peak of pkt2 now edge cancels properly
- messed up the positioning on the jab lock hitbox so i fixed it

O1yyy committed:
Merge branch 'dev' into ness-fixups

O1yyy committed:
more ness touchups

- Dtilt adjusted to P+
  - angle 361 -> 83
  - inner hitbox damage 4.5% -> 3.0%, does same damage allover
  - KBG: 3 -> 135
  - BKB: 20 -> 10
  - hitlag mult: 1.0 -> 0.85
  - sdi mult: 1.0 -> 0.5
- magnet adjusted to P+
  - angle: 351 -> 55
  - KBG: 35 -> 90
  - BKB: 70 -> 27
- decreased size of dj effect

O1yyy committed:
dthrow kbg/bkb changes + recovery adjustment

dthrow
- kbg: 85 -> 65
- bkb: 70 -> 80

up b
- decreased initial speed: 3.7 -> 3.5

added a wall jump
- decreased vanilla jump height
- passive_wall_jump_y_speed: 2.5 ->1.9

O1yyy committed:
lucas upspecial fixes

up b
- multihits
  - now interpolate properly
- id 0: (front head hitbox)
  - damage: 1.0% -> 2.0%
  - angle: 365 -> 366
  - kbg: 100 -> 130
  - fkb: 180 -> 50
- id 1: (back end hitbox)
  - damage: 2.0% -> 1.0% (encourage sdi attempts)
  - same stats as id 1
- finisher
  - matched hitbox layout to multihits
  - size of id 0: 12.5 -> 8.5

Riah321 commented:
"We'll leave it untouched"

9 commits later....
